src/Microsoft.Health.Fhir.Core/Features/Search/Expressions/ChainedExpression.cs

```diff
@@ -52,6 +55,11 @@ public class ChainedExpression : Expression
         /// </summary>
         public string TargetResourceType { get; }
 
+        /// <summary>
+        /// Get if the expression is reversed.
+        /// </summary>
+        public bool Reversed { get; }
```

Comment:
I've been wondering if it makes to sense to specify `ReverseChainedExpression` as its own class (or inherited class). Then if polymorphism can smooth out some of the conditionals below. Would mean that `TableExpressionKind.ReverseChain` might be a concept too. Thoughts?

Reply:
I think it's possible to split it out, but I'm not sure how much cleaner it would be. Most of the logic is in the SQL generation step and splitting it doesn't appear to make that any cleaner/easier. Most of the logic between chain and reverse chain is the same, just swapping what we consider `Sid1` and `Sid2` and the occasional table change so we're filtering correctly.